### PR TITLE
Greatly improve the performance of GIF processing with ImageMagick

### PR DESCRIPTION
**Pull request by dtdesign**
See https://community.woltlab.com/thread/285609-performance-engpass-bei-der-verarbeitung-von-gifs/

### CODE REVIEW

**Comment by SoftCreatR** [comment]
https://stackoverflow.com/questions/62481526/imagemagick-quantizeimage-100x-slower-on-newer-versions

**Comment by dtdesign** [comment]
Thanks, but I could not verify the findings with recent ImageMagick versions. The run difference between enabling or disabling the dither falls into the statistical error.

Also running `identify` on the sample image I've used reports 269 frames. Rescaling them within 8 seconds is perfectly fine, each frame took around 30ms. This includes unpacking the GIF by reconstructing each frame, scaling each one and putting it all back together. 8s appears to be slow, but 30ms per frame on a notebook CPU with disabled turbo is perfectly fine.

**Comment by SoftCreatR** [comment]
Alright. I just came across this, when trying to find out, what that method does.

File: wcfsetup/install/files/lib/system/image/adapter/ImagickImageAdapter.class.php [hunk 1]
```diff
@@ -356,6 +356,16 @@ public function writeImage($image, $filename) {
 			fclose($file);
 		}
 		else {
```
Comment (TimWolla) [inline wcfsetup/install/files/lib/system/image/adapter/ImagickImageAdapter.class.php:358]: Should the `if` part of the condition be handled as well?

**Review by TimWolla — APPROVED** [review]
LGTM other than that question which should result in a perfectly obvious change.

File: wcfsetup/install/files/lib/system/image/adapter/ImagickImageAdapter.class.php (hunk 1 above)
Comment (dtdesign) [reply wcfsetup/install/files/lib/system/image/adapter/ImagickImageAdapter.class.php:358]: We could simply remove that `if`, this is a 6 year old workaround for ancient PHP versions using ImageMagick 3.1.0RC1 from early 2012.

Comment (TimWolla) [reply wcfsetup/install/files/lib/system/image/adapter/ImagickImageAdapter.class.php:358]: I'd be fine with that.

**Review by TimWolla — APPROVED** [review]

**Pull request merged**